### PR TITLE
Add dockerfile

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+FROM rust:1-slim-stretch as builder
+
+WORKDIR /srv/addresses-importer
+
+ENV DEBIAN_FRONTEND noninteractive
+RUN apt-get update && apt-get install -y gcc-4.8 && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+ENV CC "gcc-4.8"
```
Comment (remi-dupre) [inline Dockerfile]: Do we need to freeze a compiler version? (I don't know about C usages, but I often see this, I guess there is a reason?)

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
+FROM rust:1-slim-stretch as builder
+
+WORKDIR /srv/addresses-importer
+
+ENV DEBIAN_FRONTEND noninteractive
+RUN apt-get update && apt-get install -y gcc-4.8 && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
```
Comment (remi-dupre) [inline Dockerfile]: Some [libpostal dependancies](https://github.com/openvenues/libpostal#installation-maclinux) appear to be missing from the docker image. Be careful as I think libpostal omitted `libgeos` in its dependancies.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ENV CC "gcc-4.8"
+RUN git clone https://github.com/openvenues/libpostal \
+    && cd postal \
```
Comment (remi-dupre) [inline Dockerfile]:
```suggestion
    && cd libpostal \
```

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    && ./bootstrap.sh \
+    && ./configure --datadir=$(pwd)/data \
+    && make install \
```
Comment (remi-dupre) [inline Dockerfile]: I didn't know that `make install` triggers a build :)

Maybe we still want to explicitly run a `make -jX` (X = 4?) to speed things up?

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+RUN cd deduplicator && cargo build --release
+
+FROM debian:stretch-slim
+
+WORKDIR /srv
+
+ENV DEBIAN_FRONTEND noninteractive
+RUN apt-get update && apt-get install -y gcc-4.8 && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+COPY --from=builder /srv/addresses-importer/deduplicator/target/release/deduplicator /usr/bin/deduplicator
```
Comment (remi-dupre) [inline Dockerfile:44]: If I understand well, we are now exporting necessary files in a newly created docker image. Won't we miss a part of dependencies here? At least libpostal and its metadata and some of its dependencies?

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+COPY . ./
+
+RUN cd deduplicator && cargo build --release
+
+FROM debian:stretch-slim
+
+WORKDIR /srv
+
+ENV DEBIAN_FRONTEND noninteractive
+RUN apt-get update && apt-get install -y gcc-4.8 && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
```
Comment (remi-dupre) [inline Dockerfile]: Why do we need `gcc` in the final image?

**Review by remi-dupre — CHANGES_REQUESTED** [review]
When building the image, it appears that docker starts by loading all content of current directory. It would be helpful to skip large files that could appear in dev workflow by adding a `.dockerignore`:

```
**/addresses.db
**/target
```

File: Dockerfile (hunk 1 above)
Comment (GuillaumeGomez) [reply Dockerfile]: This is what is being used for compiling libpostal. So maybe not necessary but not sure either...

**Comment by GuillaumeGomez** [comment]
Needs to add libgeos dependency to finish the deduplicator build.

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+RUN git clone https://github.com/openvenues/libpostal libpostal \
+    && cd libpostal \
+    && git checkout ${LIBPOSTAL_COMMIT} \
+    && ./bootstrap.sh \
+    && ./configure --datadir=/srv/data \
+    && make install -j 4 \
+    && cd .. \
+    && rm -rf libpostal \
+    && cp /usr/local/lib/libpostal.so .
```
Comment (remi-dupre) [inline Dockerfile]: Is this copy used? (A symlink would also probably be better)

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+RUN cd deduplicator && cargo build --release
+RUN rm /usr/local/lib/libpostal.so
+
+
+FROM libpostal
+
+WORKDIR /srv
+
+ENV DEBIAN_FRONTEND noninteractive
```
Comment (remi-dupre) [inline Dockerfile:42]: I think this environment variable is already relevant when `apt` is run (avoiding asking for any input by using defaults everywhere):

https://askubuntu.com/questions/972516/debian-frontend-environment-variable#972528

**Review by remi-dupre — APPROVED** [review]

**Review by amatissart — APPROVED** [review]

**Pull request merged**